maven-core/src/main/java/org/apache/maven/plugin/internal/ValidatingConfigurationListener.java

```diff
@@ -94,4 +110,57 @@ private void notify( String fieldName, Object value )
         }
     }
 
+    private void warnDeprecated( String fieldName, Object value )
+    {
+        Parameter parameter = mojoDescriptor.getParameterMap().get( fieldName );
+        String deprecated = parameter.getDeprecated();
+        if ( deprecated != null && !deprecated.isEmpty() )
+        {
+            if ( !toString( value ).equals( toString( parameter.getDefaultValue() ) ) )
+            {
+                StringBuilder sb = new StringBuilder( "  Parameter '" );
+                sb.append( fieldName ).append( '\'' );
+                if ( parameter.getExpression() != null )
+                {
+                    String userParam = parameter.getExpression().replace( "${", "'" ).replace( '}', '\'' );
+                    sb.append( " (User Parameter " ).append( userParam ).append( ")" );
```

Comment:
It is called User Property, not User Parameter.
Also be aware that the difference between expression and default-value wasn't that clear in the past, so there are still plugins around with odd values. So only show it in case of a valid expression.

Reply:
Do you have and example of such an old plugin? How do I know if it is a "valid expression" at this stage of the plugin execution?